Cabal/src/Distribution/PackageDescription/Check/Common.hs

```diff
@@ -116,34 +122,58 @@ partitionDeps ads ns ds = do
 -- for important dependencies like base).
 checkPVP
   :: Monad m
-  => (String -> PackageCheck) -- Warn message dependend on name
+  => (Dependency -> Bool)
```

Comment:
Why are we passing a new function?

Reply:
Previously the `(Dependency -> Bool)` was supplied internally as `withoutUpper :: Dependency -> Bool`. I wanted more than one of these predicates, one for each added check, so surfaced this as an argument to `checkPVP`.

Reply:
@ffaf1 Do you mean that you are looking for a way to test the performance of `cabal check`? I don't know of any tests that stress that specifically. `mem-use-tests` only tests for some very specific types of space leaks in the solver, using a relatively small number of packages.

Reply:
That `mem-use-tests` creates deep — twenty `Cond` deep iirc — `GenericPackageDescription` trees.

If you are not careful, CI won't finish on some machines (Windows). It is not `mem-use-tests` goal, but it is a useful side effect to stress test `cabal check` performance a bit!

Reply:
I was thinking of the basic test, but now I see that the later flag tests create deep conditional trees, like this one: https://github.com/haskell/cabal/blob/1103d01101a1e23c306ab788f46f377e96b2bb94/cabal-install/tests/UnitTests/Distribution/Solver/Modular/MemoryUsage.hs#L152

I didn't realize that it caused a problem for Windows CI. Do you know if there is an issue for that?

Reply:
> I didn't realize that it caused a problem for Windows CI. Do you know if there is an issue for that?

The test is perfectly fine. Not only that, it is thoroughly commented, which is a bonus.

That specific test created issues to me in the past becuase the code I wrote was not efficient. It was more evident in Win CI because I believe Win CI machine is less spec'd or otherwise performant than Linux/Mac CIs.

As it is useful and appropriate, the test stays!